### ISSUE TITLE
Trigger JS callback on data table edit

### ISSUE THREAD

**Issue by d-que** [bokeh/bokeh#9911]
Bases on [this](https://discourse.bokeh.org/t/trigger-js-callback-on-data-table-edit/5150/2), it might be good to check how it should work to trigger a (js) callback upon changing data in a DataTable. 

Currently, `data_source.js_on_event('patching', callback)` does the job. But it might feel more natural to have `data_source.js_on_event('patching', callback)` or data_table.js_on_event('edit', callback)` triggering events.

**Comment by bryevdv** [comment]
@p-himik do you have any opnions about this?

**Comment by p-himik** [comment]
I don't think this is actually DataTable-specific. A more general solution is possible that would allow users to react to _any_ `ColumnDataSource.data` value changes. Something like `data_source.js_on_event('data_changed', callback)` or something like that.
